.github/workflows/pre-commit-autoupdate.yml

```diff
@@ -0,0 +1,79 @@
+name: "Run pre-commit autoupdate"
+
+on:
+  workflow_call:
+    inputs:
+      target:
+        description: "The branch to issue the PR against"
+        type: "string"
+        default: "main"
+    secrets:
+      TOKEN:
+        description: "Your token"
+        required: true
+
+permissions:
+  contents: write
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  precommit-autoupdate:
+    name: "pre-commit autoupdate"
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Check out repository"
+        uses: actions/checkout@v4
+
+      - name: "Create a new branch"
+        id: branch
+        run: |
+          branch=pre-commit/$(date +'%Y-%m-%dT%H_%M_%S')
+          git checkout -b $branch
+          echo "name=$branch" >> $GITHUB_OUTPUT
+
+      - name: "Add brew to the PATH"
+        run: echo "/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin" >> $GITHUB_PATH
+
+      - name: "Install pre-commit"
+        run: brew install pre-commit
+
+      - name: "Check for pre-commit hook updates"
+        id: pre-commit
+        run: |
+          pre-commit autoupdate
+          updates=$(git diff --raw)
+          echo "updates=$updates" >> $GITHUB_OUTPUT
+
+      - name: "Commit local changes"
+        if: ${{ steps.pre-commit.outputs.updates }}
+        run: |
+          git config user.name "$USER"
+          git config user.email "$EMAIL"
+          git status
+          git add .
+          git commit -m "$MESSAGE"
+        env:
+          USER: "Github Build Bot"
+          EMAIL: "buildbot@fishtownanalytics.com"
+          MESSAGE: "[automated] update pre-commit hooks"
+
+      - name: "Open pull request"
+        id: pull_request
+        run: |
+          git push --set-upstream origin $HEAD
+          url=$(gh pr create -B $TARGET -H $HEAD -t "$TITLE" -b "$BODY")
+          echo "url=$url" >> $GITHUB_OUTPUT
+        env:
+          GH_TOKEN: ${{ secrets.TOKEN }}
+          TITLE: "[Automated] Update pre-commit hooks"
+          BODY: "Pre-commit hooks have updates that require resolution."
+          HEAD: ${{ steps.branch.outputs.name }}
+          TARGET: ${{ inputs.target }}
+
+      - name: "[Notification] Pull request opened"
+        run: |
+          message="PR opened at ${{ steps.pull_request.outputs.url }}"
+          echo "::notice title="Pull request opened": $title::$message"
```

Comment:
I recommend adding a step after this that posts to your slack pr channel.  Core has a couple workflows that open PRs and the PRs were just getting lost.  Adding them to the PR channel has been useful, we even tag in the support handle - although that specifically is less useful if core is using this as well.  You'll probably need a new webhook for it - you could pass it in as another secret and then if/when core starts using this pattern we can just hook into everything you've written.

Reply:
Good point. I'll add that.